java/bulk-export/bulk-export-task-execution/src/main/java/sleeper/bulkexport/taskexecution/SqsBulkExportQueueHandler.java

```diff
@@ -61,30 +62,30 @@ public SqsBulkExportQueueHandler(AmazonSQS sqsClient, TablePropertiesProvider ta
      * Receive a BulkExportLeafPartitionQuery message from the SQS queue and returns
      * a SqsMessageHandle that contains the deserialised message.
      *
-     * @return an Optional containing the SqsMessageHandle if a message was
-     *         received, or an empty Optional if no message was received
+     * @return             an Optional containing the SqsMessageHandle if a message was
+     *                     received, or an empty Optional if no message was received
      * @throws IOException if there was an error deserialising the message
      */
     public Optional<SqsMessageHandle> receiveMessage() throws IOException {
         int waitTimeSeconds = instanceProperties.getInt(BULK_EXPORT_TASK_WAIT_TIME_IN_SECONDS);
         int keepAliveFrequency = instanceProperties.getInt(BULK_EXPORT_KEEP_ALIVE_PERIOD_IN_SECONDS);
         String sqsJobQueueUrl = instanceProperties.get(LEAF_PARTITION_BULK_EXPORT_QUEUE_URL);
 
-        ReceiveMessageRequest receiveMessageRequest = new ReceiveMessageRequest(sqsJobQueueUrl)
-                .withMaxNumberOfMessages(1)
-                .withWaitTimeSeconds(waitTimeSeconds); // Must be >= 0 and <= 20
-        ReceiveMessageResult receiveMessageResult = sqsClient.receiveMessage(receiveMessageRequest);
-        if (receiveMessageResult.getMessages().isEmpty()) {
+        ReceiveMessageRequest receiveMessageRequest = ReceiveMessageRequest.builder().queueUrl(sqsJobQueueUrl)
+                .maxNumberOfMessages(1)
+                .waitTimeSeconds(waitTimeSeconds).build(); // Must be >= 0 and <= 20
+        ReceiveMessageResponse receiveMessageResult = sqsClient.receiveMessage(receiveMessageRequest);
+        if (!receiveMessageResult.hasMessages()) {
```

Comment:
The `hasMessages` method checks whether the messages field was included in the result, not whether there were any messages or not. You need to call `receiveMessageResult.messages().isEmpty()`.

Could you add a test for when this is invoked with no messages on the queue please? It looks like that test would fail at the moment.